### PR TITLE
Split simulation time provider + minor tweaks

### PR DESCRIPTION
**Pull request by sdkrach**
Pulled simulation time query into dedicated interface to be used downstream. Updated syntax to Java > 4.

Added a new simulation entity which simplifies the implementation of a simple event-based use case (as e. g. UsageEvolvers in SimuLizar)

Changes should be compatible.

### CODE REVIEW

File: bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/SimCondition.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
  * @param <M>
  *            the type of the simulation model
  */
+@FunctionalInterface
 public interface SimCondition {
```
Comment (joehen) [inline bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/SimCondition.java:18]: Any reason why we don't have an I in the name?

File: bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/ISimulationTimeProvider.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package de.uka.ipd.sdq.simulation.abstractsimengine;
+
+public interface ISimulationTimeProvider {
```
Comment (joehen) [inline bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/ISimulationTimeProvider.java:11]: some comment would be nice

File: bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/ISimRunnable.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package de.uka.ipd.sdq.simulation.abstractsimengine;
+
+@FunctionalInterface
+public interface ISimRunnable<E extends IEntity> {
```
Comment (joehen) [inline bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/ISimRunnable.java:15]: Some comment would be nice

File: bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/SimpleEventBasedSimEntity.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * A simplified version of {@link AbstractSimEntityDelegator}.
+ * 
+ * The entity supports re-occuring schedules. After being activated, the entityRouting is executed
+ * at the specified initialDelay. If the entity should be rescheduled, the entityRoutine needs to
+ * return the amount of time to the next occurence.
+ * 
+ * @author Sebastian Krach
+ */
+public abstract class SimpleEventBasedSimEntity implements IEntity {
```
Comment (joehen) [inline bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/SimpleEventBasedSimEntity.java:14]: do we need an additional test case for this class?

**Review by joehen — CHANGES_REQUESTED** [review]
At least the new interfaces should have some doc comments :)

File: bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/SimCondition.java (hunk 1 above)
Comment (sdkrach) [reply bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/SimCondition.java:18]: I can't tell you. We are very inconsistent in this.

File: bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/ISimulationTimeProvider.java (hunk 2 above)
Comment (sdkrach) [reply bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/ISimulationTimeProvider.java:11]: done.

File: bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/ISimRunnable.java (hunk 3 above)
Comment (sdkrach) [reply bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/ISimRunnable.java:15]: Done.

File: bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/SimpleEventBasedSimEntity.java (hunk 4 above)
Comment (sdkrach) [reply bundles/de.uka.ipd.sdq.simulation.abstractsimengine/src/de/uka/ipd/sdq/simulation/abstractsimengine/SimpleEventBasedSimEntity.java:14]: As we do not have any tests executed right now (which we should reconsider), i forewent creating an additional test case.

**Review by joehen — APPROVED** [review]

**Pull request merged**